### ISSUE TITLE
Re-registering of already registered username results in a 500 Internal Server Error

### ISSUE THREAD

**Issue by isc-gh** [Cardsity/issue-tracker#3]
**Describe the bug**
Re-registering of already registered username results in a 500 Internal Server Error
Console output:
```
2021/01/06 17:40:19 /home/ubuntu/go/src/github.com/Cardsity/management-api/db/repositories/gorm/base.go:45 ERROR: duplicate key value violates unique constraint "users_username_key" (SQLSTATE 23505)
[1.031ms] [rows:0] INSERT INTO "users" ("username","password","admin","role") VALUES ('sas','$argon2id$v=19$m=65536,t=1,p=4$miGlg4/WEp5I6XfwrUfTGw$DQLFbtpn157J837DfexxJSQM7hDhuf65kI7Ut66cQvs',false,'') RETURNING "id"
[GIN] 2021/01/06 - 17:40:19 | 500 |   62.317694ms |       127.0.0.1 | POST     "/v1/auth/register"
```

**To Reproduce**
1. Properly Set up API
2. Register username once (200 OK)
3. Register same username again
    -> 500 Internal Server Error

**Expected behavior**
Status code: 409 Conflict
Response body:
```
{
  "err": true,
  "errors": ["ERR_BAD_REQUEST", "ERR_USER_EXISTS"]
}
```

**Additional context**
Occurs in @Cardsity/management-api

**Comment by ravi0lii** [comment]
The commit I pushed should work as a workaround, but is kind of hacky. If there is another option that would not just be a workaround - a better workaround would work, too - I'm open for comments. Of course, no workaround would be the better option. For more explanation why I went this path, see the comment I added above the workaround: https://github.com/Cardsity/management-api/commit/0f828a44b340ec244f3c004632e51a7267135c55#diff-b2925ffbf111fb573ad2d611a5c31aa748593bba051395d7a8e0820e503cbd28R46
